scipy/signal/waveforms.py

```diff
@@ -479,3 +480,78 @@ def _sweep_poly_phase(t, poly):
     intpoly = polyint(poly)
     phase = 2 * pi * polyval(intpoly, t)
     return phase
+
+
+def unit_impulse(shape, idx=None, dtype=float):
+    """
+    Unit impulse signal (discrete delta function) or unit basis vector.
+
+    Parameters
+    ----------
+    shape : int or tuple of int
+        Number of samples in the (1D) output, or a tuple that represents the
+        shape of the output.
+    idx : float or str or None, optional
+        Index at which the value is 1.  If None, defaults to the 0th element.
+        If ``idx='mid'``, the impulse will be centered at ``shape // 2`` in
+        all dimensions.
+    dtype : data-type, optional
+        The desired data-type for the array, e.g., `numpy.int8`.  Default is
+        `numpy.float64`.
+
+    Returns
+    -------
+    y : ndarray
+        Output array containing an impulse signal.
+
+    Notes
+    -----
+    .. versionadded:: 0.19.0
```

Comment:
Well I found [conflicting references for this](https://github.com/scipy/scipy/pull/4606#issuecomment-235114172), so I just left it out and described what it does in other words.